zulipterminal/config/themes.py

```diff
@@ -597,3 +602,40 @@ def theme_with_monochrome_added(theme: ThemeSpec) -> ThemeSpec:
     "zt_light": zt_light,
     "zt_blue": zt_blue,
 }
+
+
+def generate_theme(theme_name: str, color_depth: int) -> ThemeSpec:
+    theme_styles = NEW_THEMES[theme_name].STYLES
+    urwid_theme = parse_themefile(theme_styles, color_depth)
+    return urwid_theme
+
+
+def parse_themefile(
+    theme_styles: Dict[Optional[str], Tuple[Any, Any]], color_depth: int
+) -> ThemeSpec:
+    urwid_theme = []
+    for style_name, (fg, bg) in theme_styles.items():
+        fg_code16, fg_code256, fg_code24, *fg_props = fg.value.split()
+        bg_code16, bg_code256, bg_code24, *bg_props = bg.value.split()
```

Comment:
This function reads more clearly now, right?

Reply:
Yes, this is much better now.
We could also reduce `color_depth` to just `depth` to make to more cleaner?